src/PowerShellEditorServices.Engine/Services/Analysis/AnalysisService.cs

```diff
@@ -673,6 +691,232 @@ private class PowerShellResult
 
             public bool HasErrors { get; }
         }
+
+        internal async Task RunScriptDiagnosticsAsync(
+            ScriptFile[] filesToAnalyze)
+        {
+            // If there's an existing task, attempt to cancel it
+            try
+            {
+                if (s_existingRequestCancellation != null)
```

Comment:
I know this is existing code, but I suspect there is a race condition here:

- Thread 1 checks `s_existingRequestCancellation != null`
- Thread 2 also checks `s_existingRequestCancellation != null` before thread 1 has set it to `null`
- Thread 1 then sets `s_existingRequestCancellation` to `null`
- Thread 2 dereferences `s_existingRequestCancellation`

Seems like something we haven't hit though.

More likely would be where two threads both see a non-null `s_existingRequestCancellation` and proceed to overwrite it. The last one to set `s_existingRequestCancellation` wins, but the one that gets there first is never cancelled.